kernel/packages/shared/session/index.ts

```diff
@@ -124,3 +126,18 @@ export async function userAuthentified(): Promise<void> {
     })
   })
 }
+
+function removeUrlParam(paramToRemove: string) {
+  let url = window.location.href.split('?')[0] + '?'
+  let pageURL = decodeURIComponent(window.location.search.substring(1))
+  let urlVariables = pageURL.split('&')
+  let parameterName
+
+  for (let i = 0; i < urlVariables.length; i++) {
+    parameterName = urlVariables[i].split('=')
+    if (parameterName[0] !== paramToRemove) {
+      url = url + parameterName[0] + '=' + parameterName[1] + '&'
+    }
+  }
+  window.history.replaceState({}, document.title, url.substring(0, url.length - 1))
+}
```

Comment:
I think we could use this function [Location search](https://developer.mozilla.org/en-US/docs/Web/API/Location/search) and then remove elements with `params.delete(paramToRemove)`

Reply:
You're right! I have applied a refactor of the `removeUrlParam()` function. Please take a look! Thanks! :)